source/extensions/filters/http/transformation/inja_transformer.h

```diff
@@ -82,15 +84,23 @@ class Extractor : Logger::Loggable<Logger::Id::filter> {
   absl::string_view extract(Http::StreamFilterCallbacks &callbacks,
                             const Http::RequestOrResponseHeaderMap &header_map,
                             GetBodyFunc &body) const;
-
 private:
   absl::string_view extractValue(Http::StreamFilterCallbacks &callbacks,
                                  absl::string_view value) const;
+  absl::string_view replaceIndividualValue(Http::StreamFilterCallbacks &callbacks,
+                                           absl::string_view value) const;
+  absl::string_view replaceAllValues(Http::StreamFilterCallbacks &callbacks,
+                                     absl::string_view value) const;
 
   const Http::LowerCaseString headername_;
   const bool body_;
   const unsigned int group_;
   const std::regex extract_regex_;
+  const std::optional<std::string> replacement_text_;
+  const ExtractionApi::Mode mode_;
+
+  ExtractionFunc extraction_func_;
+  mutable std::string replaced_value_;
```

Comment:
as far as i can see, `replaced_value_` is only ever written to, but never read. do correct me if i'm wrong, but if not, is it necessary to keep this private variable around?

Reply:
> as far as i can see, replaced_value_ is only ever written to, but never read

This is correct and intentional. The reason for this is that the various `ExtractionFunc` methods (`extractValue`, `replaceIndividualValue`, `replaceAllValues`) return `absl::string_view`, which is essentially a reference to a `std::string`. If we try to return a `absl::string` view from one of these methods without storing the post-replace value  somewhere, the underlying `std::string` will go out of scope before we can interact with the result

e.g., if we try (updated snippet from `replaceAllValues`)
```diff
  // If a match was found, replace all instances of the regex in the input value with the replacement_text_ value
  replaced = std::regex_replace(input, extract_regex_, replacement_text_.value());

-  // Store the replaced string in replaced_value_ and return an absl::string_view to it
-  replaced_value_ = std::move(replaced);
-  return absl::string_view(replaced_value_);
+  return absl::string_view(replaced);
```

we lose the value of `replaced` as soon as the method exits, causing the output to be interpreted as an empty string, even if that is not intended

Reply:
okay, makes sense! could you just document that on the variable declaration?

Reply:
actually, i'm still a little hung up on this. why not change `ExtractionFunc` to return a string instead? or even better, a `std::unique_ptr<std::string>`? it feels a little strange to use a class variable to hold a return value on the stack.

overall, this seems like a slightly strange pattern to me, but if this is an accepted practice, i wouldn't get too hung up on it.

Reply:
> Why not change `ExtractionFunc` to return a string instead

 - The `Extractor` uses the `ExtractionFunc`s in `Extractor::extract`, which predates this PR and returns an `absl::string_view`. If I wanted to update these methods to return a `std::string` or a `std::unique_ptr<std::string>`, I would need to either:
   - Update `Extractor::extract` to return an `std::string`(or `std::unique_ptr<std::string>`), which increases the risk (increased potential for string corruption bugs) and expands the scope of this PR
     - There really are a lot of changes that would need to follow from here if we want to go this route
   - Continue returning an `absl::string_view` from `Extractor::extract`, which leaves us in the same position we are now (i.e., we need to return a `absl::string_view` pointing to a `std::string`/`std::unique_ptr<std::string>` which will go out of scope when the function exits)

Reply:
I agree, using a class variable to store the replace computation results isn't ideal. However, given how extractors currently interface with the transformation filter, it seems like the least disruptive way to handle this situation. We've got a series of steps outside of `Extractor::extract` that depend on `absl::string_view`s pointing to the replace/extract results. Tying the results to the lifetime of the `Extractor` object is a straightforward way to ensure that these `absl::string_view`s remain in scope.

Reply:
Thanks! I agree on all points. It's not ideal, but it works, and this isn't a great time to change this, so let's leave it for now. Perhaps we can consider revisiting this at some point in the future.